### PR TITLE
[Disk Manager]: Add prometheus monitoring to opensource main

### PR DESCRIPTION
**Pull request by jkuradobery**
Add prometheus metrics as an option

### CODE REVIEW

File: cloud/disk_manager/pkg/monitoring/metrics/prometheus_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package metrics
+
+import (
+	"github.com/stretchr/testify/require"
+
+	"testing"
```
Comment (SvartMetal) [inline cloud/disk_manager/pkg/monitoring/metrics/prometheus_test.go:6]: Order

File: cloud/disk_manager/pkg/monitoring/metrics/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
+package metrics
+
+import (
+	"net/http"
+	"regexp"
+	"sync"
+
+	prometheus2 "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
```
Comment (SvartMetal) [inline cloud/disk_manager/pkg/monitoring/metrics/prometheus.go:10]: ?

File: cloud/disk_manager/pkg/monitoring/metrics/prometheus.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+)
+
+////////////////////////////////////////////////////////////////////////////////
+
+type gatherer struct {
+	sync.Mutex
+	gatherers prometheus2.Gatherers
+}
+
+func (r *gatherer) Add(registry prometheus2.Gatherer) {
```
[the remark this answers is not in the document]
Comment (SvartMetal) [reply cloud/disk_manager/pkg/monitoring/metrics/prometheus.go:23]: gatherer prometheus2.Gatherer

File: cloud/disk_manager/pkg/monitoring/metrics/prometheus_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+////////////////////////////////////////////////////////////////////////////////
+
+func TestSanitizeName(t *testing.T) {
```
Comment (SvartMetal) [inline cloud/disk_manager/pkg/monitoring/metrics/prometheus_test.go:11]: TestPrometheusSanitizeName

**Review by SvartMetal — APPROVED** [review]

**Review by BarkovBG — APPROVED** [review]

**Review by SvartMetal — APPROVED** [review]

**Review by SvartMetal — APPROVED** [review]

**Pull request merged**